tests/ProductionConfigTests.js

```diff
@@ -24,28 +24,710 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
+                    },
+                    method: "PUT",
+                    expectedStatusCode: 404,
+                    expectedPayload: {
+                        "isError": true,
+                        "message": "Cannot update:  GPII key \"" + gpii.tests.productionConfigTesting.carlaKey + "\" is a snapset while executing HTTP PUT on url http://preferences:9081/preferences/%gpiiKey?merge=%merge"
+                    }
+                }
+            }
         }
     });
+    gpii.test.push(testDef.sequence, [
+        {
+            func: "{healthRequest}.send"
+        }, {
+            event: "{healthRequest}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testResponse"
+        }, {
+            func: "{readyRequest}.send"
+        }, {
+            event: "{readyRequest}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testResponse"
+        }, {
+            func: "{accessTokenRequest}.send",
+            args: [gpii.tests.productionConfigTesting.carlaTokenRequestPayload]
+        }, {
+            event: "{accessTokenRequest}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testAccessResponse"
+        }, {
+            func: "{lifeCycleRequest}.send",
+            args: [
+                null,
+                {
+                    "headers": {
+                        "Authorization": "{accessTokenRequest}.options.stashedAuth"
+                    }
+                }
+            ]
+        }, {
+            event: "{lifeCycleRequest}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testLifecycleResponse"
+        }, {
+            func: "{putSettingsRequestFailure}.send",
+            args: [
+                gpii.tests.productionConfigTesting.prefsUpdate,
+                {
+                    "headers": {
+                        "Authorization": "{accessTokenRequest}.options.stashedAuth"
+                    }
+                }
+            ]
+        }, {
+            event: "{putSettingsRequestFailure}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testResponse"
+        }
+    ]);
     return testDef;
-}));
+});
+
+// Override the original "kettle.test.testDefToServerEnvironment" function provided by kettle library to boil a new
+// aggregate event "onAllReady" that listens to both "onServerReady" and "{flowManager}.events.resetAtStartSuccess" events
+kettle.test.testDefToServerEnvironment = function (testDef) {
+    var configurationName = testDef.configType || kettle.config.createDefaults(testDef.config);
+    return {
+        type: "kettle.test.serverEnvironment",
+        options: {
+            configurationName: configurationName,
+            components: {
+                tests: {
+                    options: kettle.test.testDefToCaseHolder(configurationName, testDef)
+                }
+            },
+            events: {
```

Comment:
This "events" block is no longer needed too with the merge for GPII-3424 pull request. Please also fix the comment at line 379. Thanks.

Reply:
In that case, this override of `kettle.test.testDefToServerEnvironment()` collapses to the one defined in`KettleTestUtils.js`.  I'll remove the definition entirely from here.